test/e2e/route_advertisements.go

```diff
@@ -463,3 +472,397 @@ spec:
+				}),
+			ginkgo.Entry("pod in the default network should not be able to access a UDN service",
+				func(ipIndex int) (clientName string, clientNamespace string, dst string, expectedOutput string, expectErr bool) {
+					return podNetDefault.Name, podNetDefault.Namespace, net.JoinHostPort(svcNetA.Spec.ClusterIPs[ipIndex], "8080") + "/clientip", curlConnectionTimeoutCode, true
+				}),
+			ginkgo.Entry("pod in the UDN should be able to access a service in the same network",
+				func(ipIndex int) (clientName string, clientNamespace string, dst string, expectedOutput string, expectErr bool) {
+					return podsNetA[0].Name, podsNetA[0].Namespace, net.JoinHostPort(svcNetA.Spec.ClusterIPs[ipIndex], "8080") + "/clientip", "", false
+				}),
+			ginkgo.Entry("pod in the UDN should not be able to access a default network service",
+				func(ipIndex int) (clientName string, clientNamespace string, dst string, expectedOutput string, expectErr bool) {
+					err := true
+					out := curlConnectionTimeoutCode
+					if isLocalGWModeEnabled() {
+						// FIXME: L3 - pod in the UDN should NOT be able to access a default network service
+						err = false
+						out = ""
+					}
+					return podsNetA[0].Name, podsNetA[0].Namespace, net.JoinHostPort(svcNetDefault.Spec.ClusterIPs[ipIndex], "8080") + "/clientip", out, err
+				}),
+			ginkgo.Entry("pod in the UDN should not be able to access a service in a different UDN",
+				func(ipIndex int) (clientName string, clientNamespace string, dst string, expectedOutput string, expectErr bool) {
+					return podsNetA[0].Name, podsNetA[0].Namespace, net.JoinHostPort(svcNetB.Spec.ClusterIPs[ipIndex], "8080") + "/clientip", curlConnectionTimeoutCode, true
+				}),
+			ginkgo.Entry("host to a local UDN pod should not work",
+				func(ipIndex int) (clientName string, clientNamespace string, dst string, expectedOutput string, expectErr bool) {
+					statusPodA, err := userDefinedNetworkStatus(podsNetA[0], namespacedName(podsNetA[0].Namespace, cudnATemplate.Name))
+					framework.ExpectNoError(err)
+					// FIXME: L3 - curl returns code 7, the request works partially
+					return podsNetA[0].Spec.NodeName, "", net.JoinHostPort(statusPodA.IPs[ipIndex].IP.String(), "8080") + "/clientip", "", true
+				}),
+			ginkgo.Entry("host to an external UDN pod should not work",
+				func(ipIndex int) (clientName string, clientNamespace string, dst string, expectedOutput string, expectErr bool) {
+					// podsNetA[0] and podsNetA[2] are on different nodes
+					statusPodA, err := userDefinedNetworkStatus(podsNetA[0], namespacedName(podsNetA[0].Namespace, cudnATemplate.Name))
+					framework.ExpectNoError(err)
+					// FIXME: L3 - host to UDN pod on different node should not work
+					return podsNetA[2].Spec.NodeName, "", net.JoinHostPort(statusPodA.IPs[ipIndex].IP.String(), "8080") + "/clientip", "", false
+				}),
+		)
+
+	},
+	ginkgo.Entry("Layer3",
```

Comment:
nit: maybe update the PR title/commit message to indicate this is for layer3 only and layer 2 will be added later?